tests/cloud_tests/platforms/azurecloud/platform.py

```diff
@@ -111,6 +111,14 @@ def get_image(self, img_conf):
 
     def destroy(self):
         """Delete all resources in resource group."""
+        preserve_instance = __import__(
+            'tests.cloud_tests.stage').PlatformComponent.preserve_instance
```

Comment:
As we discussed in IRC, this won't work as you need the PlatformComonent instance that's created in collect.py.  I suggest that we'll need to pull in the args.preserve_instance flag will need be passed to the platform object:

collect.py:collect() takes args, and passes args to collect.py:collect_platform(), and we may need to pull out values from args and put them into platform_config.

Reply:
That's a good idea if we pull the args from collect py and put them in platform configs, we can use the platform config in platforms to stop the deletion of the resource. Where should I put this, since this will also affect other clouds?

Reply:
Since it's only config; unless the other platforms look for the key in the config; it won't have an impact.

```
% git diff
diff --git a/tests/cloud_tests/collect.py b/tests/cloud_tests/collect.py
index 642745d..84a6009 100644
--- a/tests/cloud_tests/collect.py
+++ b/tests/cloud_tests/collect.py
@@ -187,6 +187,8 @@ def collect_platform(args, platform_name):
     platform_config = config.load_platform_config(
         platform_name, require_enabled=True)
     platform_config['data_dir'] = args.data_dir
+    platform_config['preserve_instance'] = args.preserve_instance
+    platform_config['preserve_data'] = args.preserve_data
     LOG.debug('platform config: %s', platform_config)
     component = PlatformComponent(
         partial(platforms.get_platform, platform_name, platform_config))
diff --git a/tests/cloud_tests/platforms/azurecloud/platform.py b/tests/cloud_tests/platforms/azurecloud/platform.py
index cb62a74..6fb9740 100644
--- a/tests/cloud_tests/platforms/azurecloud/platform.py
+++ b/tests/cloud_tests/platforms/azurecloud/platform.py
@@ -40,6 +40,8 @@ class AzureCloudPlatform(Platform):
         self.storage_sku = config['storage_sku']
         self.vm_size = config['vm_size']
         self.location = config['region']
+        self.preserve_instance = config['preserve_instance']
```